ehr_sqlite3.py

```diff
@@ -0,0 +1,188 @@
+# a patient class that stores all the information.
+def create_database(labpath: str, patientpath: str, dbpath: str) -> None:
+    """Extract information from txt file."""
+    con = sqlite3.connect(dbpath)
+    cur = con.cursor()
+
+    cur.execute(
+        """CREATE TABLE Patients(\
+            [PatientID] TEXT PRIMARY KEY,\
+            [PatientGender] TEXT,\
+            [PatientDateofBirth] TEXT,\
+            [PatientRace] TEXT,\
+            [PatientMaritalStatus] TEXT,\
+            [PatientLanguage] TEXT,\
+            [PatientPopulationPercentageBelowPoverty] TEXT\
+            )"""
+    )
+
+    cur.execute(
+        """CREATE TABLE labs(\
+            [PatientID] TEXT NOT NULL,\
+            [AdmissionID] TEXT NOT NULL,\
+            [LabName] TEXT,\
+            [LabValue] TEXT,\
+            [LabUnits] TEXT,\
+            [LabDateTime] TEXT\
+            )"""
+    )
+
+    with open(patientpath, encoding="UTF-8-sig") as file:
+        counter: int = 0
+
+        for line in file:
+            counter += 1
+            line1: list[str] = line.split("\t")
+            line1[-1] = line[-1][:-1]
+
+            if counter == 1:
+                pass
+            else:
+                cur.execute("INSERT INTO Patients values (?,?,?,?,?,?,?)", line1)
+
+    with open(labpath, encoding="UTF-8-sig") as file:
+        counter2: int = 0
+
+        for line in file:
+            counter2 += 1
+            line2: list[str] = line.split("\t")
+            line2[-1] = line2[-1][:-1]
+
+            if counter2 == 1:
+                pass
+            else:
+                cur.execute("INSERT INTO Labs values (?,?,?,?,?,?)", line2)
+    cur.close()
+
+
+# Below we define a lab class
+# This lab class takes lab name and patientID
+# to make a lab class
+# we take into account the lab and patient
+# becasue we do not have a unique identifier in
+# our lab class.
+
+
+class Lab:
+    """Store Lab data."""
+
+    def __init__(self, name: str, patientID: str, dbpath: str) -> None:
+        """Instantiate a class."""
+        self.name: str = name
+        self.patientID: str = patientID
+        con = sqlite3.connect(dbpath)
+        self.cur = con.cursor()
+
+    def __str__(self) -> str:
+        """Lab Class Representation."""
+        return str(self.name) + " attended by " + str(self.patientID)  # O(1)
+
+    @property
+    def units(self) -> str:
+        """Return lab units."""
+        unit: tuple = self.cur.execute(
+            """select LabUnits from Labs where LabName = ? and PatientID = ?""",
+            (
+                self.name,
+                self.patientID,
+            ),
+        ).fetchone()
+        return unit[0]
+
+    @property
+    def value(self) -> float:
+        """Return lab units."""
+        value: tuple[str] = self.cur.execute(
+            """select LabValue from Labs where LabName = ? and PatientID = ?""",
+            (self.name, self.patientID),
+        ).fetchone()
+        return float(value[0])
```

Comment:
You could also cast this to a float on the way into the database and make the column type REAL. That could be faster and would allow you to do things like numeric comparisons directly in the SQL.